.github/workflows/build-stitching.yml

```diff
@@ -0,0 +1,66 @@
+name: Run tests
+on:
+  push:
+    branches:
+      - develop
+      - main
+  pull_request:
+    branches:
+      - develop
+      - main
+
+jobs:
+  build:
+    name: Run tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    steps:
+    - uses: actions/checkout@main
+      with:
+        repository: biomedia-mira/stitching   
+        ref: ${{ github.event.inputs.git_ref || 'd31242797e0438c5de7f5fc69b14697f71c090d5' }} 
+
+    - name: prepare_system
+      run: |
+        ROOT=$(pwd)
+        # prepare 3-party libs
+        export THIRD_PARTY_DIR=$ROOT/3rdParty
+        mkdir $THIRD_PARTY_DIR
+
+        # get Eigen
+        EIGEN_VERSION="3.3.7"
+        cd $THIRD_PARTY_DIR
+        wget https://gitlab.com/libeigen/eigen/-/archive/${EIGEN_VERSION}/eigen-${EIGEN_VERSION}.tar.gz --progress=bar:force:noscroll
+        mkdir eigen
+        tar xf eigen-${EIGEN_VERSION}.tar.gz -C eigen --strip-components=1
+
+        # download and install ITK
+        ITK_VERSION="5.0.0"
+        cd $THIRD_PARTY_DIR
+        wget https://sourceforge.net/projects/itk/files/itk/${ITK_VERSION%.*}/InsightToolkit-${ITK_VERSION}.tar.gz --progress=bar:force:noscroll
+        tar xf InsightToolkit-${ITK_VERSION}.tar.gz
+        cd InsightToolkit-${ITK_VERSION}
+        mkdir build
+        cd build
+        cmake \
+            -DBUILD_EXAMPLES:BOOL=OFF \
+            -DBUILD_TESTING:BOOL=OFF \
+            -DCMAKE_INSTALL_PREFIX=$THIRD_PARTY_DIR/itk \
+          ..
+        make -j$(nproc)
+        make install
+
+    - name: get_boost_apt  
+      run: sudo apt-get install -y libboost-all-dev libtbb-dev
```

Comment:
I've been trying to build `stitching` locally without using the prebuilt apt package, and arrived at these steps:

```bash
cd $THIRD_PARTY_DIR
wget https://boostorg.jfrog.io/artifactory/main/release/1.80.0/source/boost_1_80_0.tar.gz --progress=bar:force:noscroll
tar xf boost_1_80_0.tar.gz 
cd boost_1_80_0/
mkdir build
# build Boost 
./bootstrap.sh --prefix=build/
./b2
# point cmake at the location of the various boost cmake files
export CMAKE_PREFIX_PATH=../3rdParty/boost_1_80_0/stage/lib/cmake/
```

I also had to point `stitching` at the built ITK libraries, too?

```bash
export ITK_DIR=../3rdParty/InsightToolkit-5.0.0/build
```

But apart from those steps, I was able to build `stitching` using the static Boost libraries just fine! 

```console
joshua@tadpole:~/Desktop/build_biomedia_stitching/build$ make
[ 12%] Building CXX object mia/CMakeFiles/mia.dir/src/miaHistogram.cpp.o
[ 25%] Building CXX object mia/CMakeFiles/mia.dir/src/miaImage.cpp.o
[ 37%] Building CXX object mia/CMakeFiles/mia.dir/src/miaImageProcessing.cpp.o
[ 50%] Linking CXX static library libmia.a
[ 50%] Built target mia
[ 62%] Building CXX object itkio/CMakeFiles/itkio.dir/src/itkio.cpp.o
[ 75%] Linking CXX static library libitkio.a
[ 75%] Built target itkio
[ 87%] Building CXX object stitching/CMakeFiles/stitching.dir/src/main.cpp.o
[100%] Linking CXX executable stitching
[100%] Built target stitching
```

I'll see if I can port these steps to our GitHub Actions workflow now. :slightly_smiling_face:

Reply:
I then tried running `sct_image -stitch` command from https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3865#discussion_r956085087 using this new binary, and it looks like it worked for me?

```console
joshua@tadpole:~/Desktop/build_biomedia_stitching$ cp build/stitching/stitching $SCT_DIR/bin
joshua@tadpole:~/Desktop/build_biomedia_stitching$ cd $SCT_DIR/data/sct_testing_data/t2
joshua@tadpole:~/repos/spinalcordtoolbox/data/sct_testing_data/t2$ sct_image -i sct_roi1.nii.gz sct_roi2.nii.gz -o stitched.nii.gz -stitch

--
Spinal Cord Toolbox (git-remotes/origin/jqm/3828-sct-stitching-dfb91f755620dbeca936831d2a451848c6ddd753)

sct_image -i sct_roi1.nii.gz sct_roi2.nii.gz -o stitched.nii.gz -stitch
--

/tmp/sct-20220826135409.913820-image-stitching-pg8vv26f/sct_roi1.nii.gz
cp sct_roi1.nii.gz /tmp/sct-20220826135409.913820-image-stitching-pg8vv26f/sct_roi1.nii.gz
File /tmp/sct-20220826135409.913820-image-stitching-pg8vv26f/sct_roi1.nii.gz already exists. Will overwrite it.
/tmp/sct-20220826135409.913820-image-stitching-pg8vv26f/sct_roi2.nii.gz
cp sct_roi2.nii.gz /tmp/sct-20220826135409.913820-image-stitching-pg8vv26f/sct_roi2.nii.gz
File /tmp/sct-20220826135409.913820-image-stitching-pg8vv26f/sct_roi2.nii.gz already exists. Will overwrite it.
/home/joshua/repos/spinalcordtoolbox/bin/stitching -i 'sct_roi1.nii.gz sct_roi2.nii.gz' -o /tmp/sct-20220826135409.913820-image-stitching-pg8vv26f/stitched.nii.gz -a # in /tmp/sct-20220826135409.913820-image-stitching-pg8vv26f
0 stitching image...done. took 70 ms
rm -rf /tmp/sct-20220826135409.913820-image-stitching-pg8vv26f
Generate output files...

Done! To view results, run one of the following commands (depending on your preferred viewer):
fsleyes stitched.nii.gz &

fslview_deprecated stitched.nii.gz &

fslview stitched.nii.gz &
```

Just as a sanity check (to make sure I didn't accidentally install `boost` systemwide), I switched back to the old `stitching` binary, and got this:

```console
(venv_sct) joshua@tadpole:~/repos/spinalcordtoolbox/data/sct_testing_data/t2$ sct_image -i sct_roi1.nii.gz sct_roi2.nii.gz -o stitched.nii.gz -stitch

--
Spinal Cord Toolbox (git-remotes/origin/jqm/3828-sct-stitching-dfb91f755620dbeca936831d2a451848c6ddd753)

sct_image -i sct_roi1.nii.gz sct_roi2.nii.gz -o stitched.nii.gz -stitch
--

/tmp/sct-20220826135922.332441-image-stitching-f8sq2tqz/sct_roi1.nii.gz
cp sct_roi1.nii.gz /tmp/sct-20220826135922.332441-image-stitching-f8sq2tqz/sct_roi1.nii.gz
File /tmp/sct-20220826135922.332441-image-stitching-f8sq2tqz/sct_roi1.nii.gz already exists. Will overwrite it.
/tmp/sct-20220826135922.332441-image-stitching-f8sq2tqz/sct_roi2.nii.gz
cp sct_roi2.nii.gz /tmp/sct-20220826135922.332441-image-stitching-f8sq2tqz/sct_roi2.nii.gz
File /tmp/sct-20220826135922.332441-image-stitching-f8sq2tqz/sct_roi2.nii.gz already exists. Will overwrite it.
/home/joshua/repos/spinalcordtoolbox/bin/stitching -i 'sct_roi1.nii.gz sct_roi2.nii.gz' -o /tmp/sct-20220826135922.332441-image-stitching-f8sq2tqz/stitched.nii.gz -a # in /tmp/sct-20220826135922.332441-image-stitching-f8sq2tqz
Traceback (most recent call last):
  File "/home/joshua/repos/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_image.py", line 607, in <module>
    main(sys.argv[1:])
  File "/home/joshua/repos/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_image.py", line 361, in main
    im_out = stitch_images(fnames_in=fnames_in)
  File "/home/joshua/repos/spinalcordtoolbox/spinalcordtoolbox/image.py", line 1724, in stitch_images
    status, output = run_proc(cmd, verbose='verbose', is_sct_binary=True)
  File "/home/joshua/repos/spinalcordtoolbox/spinalcordtoolbox/utils/sys.py", line 414, in run_proc
    raise RuntimeError(output)
RuntimeError: /home/joshua/repos/spinalcordtoolbox/bin/stitching: error while loading shared libraries: libboost_program_options.so.1.71.0: cannot open shared object file: No such file or directory
```

:D

Reply:
Excellent work @joshuacwnewton ! :+1:

Reply:
@joshuacwnewton do you want to push the changes :) ?

Reply:
I'm happy to take care of this, yes! :)